server/src/main/kotlin/net/horizonsend/ion/server/generation/generators/OreGenerator.kt

```diff
@@ -0,0 +1,144 @@
+package net.horizonsend.ion.server.generation.generators
+
+import net.horizonsend.ion.common.loadConfiguration
+import net.horizonsend.ion.server.IonServer
+import net.horizonsend.ion.server.NamespacedKeys
+import net.horizonsend.ion.server.generation.PlacedOre
+import net.horizonsend.ion.server.generation.PlacedOres
+import net.horizonsend.ion.server.generation.PlacedOresDataType
+import net.horizonsend.ion.server.generation.configuration.AsteroidConfiguration
+import net.horizonsend.ion.server.generation.configuration.Ore
+import net.minecraft.util.RandomSource
+import net.minecraft.world.level.chunk.LevelChunkSection
+import net.starlegacy.util.nms
+import org.bukkit.Bukkit.createBlockData
+import org.bukkit.Chunk
+import org.bukkit.Material
+import org.bukkit.World
+import org.bukkit.block.data.BlockData
+
+object OreGenerator {
+	// default asteroid configuration values
+	private val configuration: AsteroidConfiguration =
+		loadConfiguration(IonServer.Ion.dataFolder.resolve("asteroids"), "asteroid_configuration.conf")
+
+	val weightedOres = oreWeights()
+	val asteroidBlocks: MutableSet<Material> = mutableSetOf()
+	private val oreMap: MutableMap<String, BlockData> = mutableMapOf()
+
+	init {
```

Comment:
is this init block even needed? just init the variables using a .map {} op

Reply:
For the materials set, yes, since its adding multiple values per entry. Unless theres an easier way of doing it. Can replace the oreMap with a .map {} though

Reply:
Map to array of arrays and join the arrays